src/screens/Exchange/utils.js

```diff
@@ -63,3 +103,190 @@ export const getAvailable = (_min: string, _max: string, rate: string) => {
   return (new BigNumber(askRate)).multipliedBy(amountToSell).toFixed();
 };
+
+export const getBestAmountToBuy = (offers: Offer[], fromAmount: string): ?string => {
+  const bestRate = maxBy(offers, 'askRate')?.askRate;
+  if (!bestRate || !+fromAmount) return null;
+  return calculateAmountToBuy(bestRate, fromAmount);
+};
+
+export const getFormattedSellMax = (asset: Option): string => {
+  const { assetBalance, symbol, formattedBalanceInFiat } = asset;
+  if (!(assetBalance && symbol && formattedBalanceInFiat)) return '';
+  return `${formatAmount(assetBalance, 2)} ${symbol} (${formattedBalanceInFiat.replace(' ', '')})`;
+};
+
+export const validateInput = (
+  fromAmount: string,
+  fromAsset: ?Option,
+  toAsset: ?Option,
+  errorMessage: string,
+): boolean =>
+  !errorMessage && !!+fromAmount && fromAmount[fromAmount.length - 1] !== '.' && !!fromAsset && !!toAsset;
+
+const generateAssetsOptions = (
+  assets: Assets,
+  exchangeSupportedAssets: Asset[],
+  balances: Balances,
+  baseFiatCurrency: ?string,
+  rates: Rates,
+): Option[] => {
+  return sortAssets(assets)
+    .filter(({ symbol }) => (getBalance(balances, symbol) !== 0 || symbol === ETH)
+      && !!exchangeSupportedAssets.some(asset => asset.symbol === symbol))
+    .map(({ symbol, iconUrl, ...rest }) => {
+      const assetBalance = formatAmount(getBalance(balances, symbol));
+      const formattedBalanceInFiat = getFormattedBalanceInFiat(baseFiatCurrency, assetBalance, rates, symbol);
+      const imageUrl = iconUrl ? `${SDK_PROVIDER}/${iconUrl}?size=3` : '';
+
+      return ({
+        key: symbol,
+        value: symbol,
+        imageUrl,
+        icon: iconUrl,
+        iconUrl,
+        symbol,
+        ...rest,
+        assetBalance,
+        formattedBalanceInFiat,
+        customProps: {
+          balance: !!formattedBalanceInFiat && {
+            balance: assetBalance,
+            value: formattedBalanceInFiat,
+            token: symbol,
+          },
+          rightColumnInnerStyle: { alignItems: 'flex-end' },
+        },
+      });
+    });
+};
+
+const generateSupportedAssetsOptions = (
+  exchangeSupportedAssets: Asset[],
+  balances: Balances,
+  baseFiatCurrency: ?string,
+  rates: Rates,
+): Option[] => {
+  if (!Array.isArray(exchangeSupportedAssets)) return [];
+  return exchangeSupportedAssets
+    .map(({ symbol, iconUrl, ...rest }) => {
+      const rawAssetBalance = getBalance(balances, symbol);
+      const assetBalance = rawAssetBalance ? formatAmount(rawAssetBalance) : '';
+      const formattedBalanceInFiat = getFormattedBalanceInFiat(baseFiatCurrency, assetBalance, rates, symbol);
+      const imageUrl = iconUrl ? `${SDK_PROVIDER}/${iconUrl}?size=3` : '';
+
+      return {
+        key: symbol,
+        value: symbol,
+        icon: iconUrl,
+        imageUrl,
+        iconUrl,
+        symbol,
+        ...rest,
+        assetBalance,
+        formattedBalanceInFiat,
+        customProps: {
+          balance: !!formattedBalanceInFiat && {
```

Comment:
`customProps.balance` can come as `false`, is that expected?

Reply:
I've just noticed `customProps.balance` isn't used anymore, so I'll just delete it